[DealabsBride] Fix for the new site

sysadminstory opened the pull request:
Dealabs changed his page structure
* New Website structure handled
* Loss of functionnality (no more category, expired deal or local deal
filter)

sysadminstory committed:
[DealabsBride] Fix for the new site

Dealabs changed his page structure
* New Website structure handled
* Loss of functionnality (no more category, expired deal or local deal
filter)

sysadminstory committed:
Coding policy conformance

sysadminstory committed:
Coding policy conformance

sysadminstory commented:
This fixes the issue #594 !

The new website is a nightmare to parse, but I did what I could.

This leads to a loss of functionality which is almost the same as on the original website

sysadminstory committed:
Merge remote-tracking branch 'upstream/master'

teromene commented on bridges/DealabsBridge.php:
```diff
@@ -110,24 +28,171 @@ public function collectData(){
+	{
+		if($deal->find(
+			'span[class*=thread-price text--b vAlign--all-tt cept-tp '
+			. 'size--all-m size--fromW2-xxl size--fromW4-xxxl]', 0) != null) {
```
No need to concatenate here, just use \ followed by a line return

teromene commented on bridges/DealabsBridge.php:
```diff
@@ -110,24 +28,171 @@ public function collectData(){
+	{
+		if($deal->find(
+			'img[class=thread-image width--all-auto height--all-auto '
+			. 'imgFrame-img cept-thread-img img--dummy js-lazy-img]', 0) != null) {
```
same as before

teromene reviewed (changes requested):
Thanks for the fix ! Just two small changes

sysadminstory committed:
Merge remote-tracking branch 'upstream/master'

sysadminstory committed:
Dealabs changed his page structure again

New Website structure handled

sysadminstory committed:
Coding policy conformance

sysadminstory commented:
I tried both way : one full string, or string on different lines concatenated : travis does not allow it.

I do not figured how do you like I change my code using antislash : this will add some new line, and the find function would fail.

I think I need some example or advice :cold_sweat: !

logmanoriginal commented:
If you build the selector with the [implode()](http://php.net/manual/en/function.implode.php) function it splits into multiple lines that are actually more readable in my opinion.

Have a look at FB2Bridge for reference: https://github.com/RSS-Bridge/rss-bridge/blob/7939bffcdd4a07fe4e015e35c9e33d20e208421b/bridges/FB2Bridge.php#L168-L178

In your case this 

```PHP
$deal->find('img[class=thread-image width--all-auto height--all-auto imgFrame-img cept-thread-img img--dummy js-lazy-img]', 0)
```
could convert into this:

```PHP
$selector = implode(
    ' ', /* Notice this is a space! */
    array(
        "thread-image",
        "width--all-auto",
        "height--all-auto",
        "imgFrame-img",
        "cept-thread-img",
        "img--dummy",
        "js-lazy-img"
    )
);

$deal->find('img[class=' . $selector . ']', 0);
/* I had to use spaces for indentation, make sure to 
replace them with tabs if you copy this code */
```

Does that work for you?

sysadminstory committed:
Coding policy conformance

A little bit more readability in the code

sysadminstory committed:
Coding policy conformance

More fix to the code

sysadminstory commented:
I finally got it !

Thank you @LogMANOriginal !

sysadminstory committed:
Bonus features : more search parameters

Here are some user configurable search parameters
- filter expired deal
- filter local deal
- filter by price (Min, Max)

Other parameters are now used for every search :
- Keyword are searched in Titres & Descriptions & Codes
- Deals are sorted newest first (it is what RSS is for : showing newest
  things !)
- Deal search timeframe is not limited

Function has been documented.

sysadminstory committed:
Coding policy fix

teromene commented:
Merged, thanks !
